src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -738,7 +738,8 @@ public SyntaxNode Declaration(ISymbol symbol)
 
                 case SymbolKind.Property:
                     var property = (IPropertySymbol)symbol;
-                    return property.IsIndexer ? IndexerDeclaration(property) : PropertyDeclaration(property);
+                    var isIndexer = property.IsIndexer || property.ExplicitInterfaceImplementations.Any(i => i.IsIndexer);
```

Comment:
this seems odd.  i would think it was a compiler bug if a property wasnt' marked as an indexer, while it's ExplicitImpls were... we should check with the compiler team if this is expected, or if it's just a bug that was never noticed (my money is on the latter).

Reply:
Here's the discussion around that. https://github.com/dotnet/roslyn/issues/53911  I tend to agree that it's a bug.  IMO a better fix for this is in the compiler API since it doesn't require workarounds here.  I opened this draft PR to highlight what sort of workarounds are required.

Reply:
@CyrusNajmabadi how do you suggest we proceed with the compiler bug discussion?  I cannot reactivate that issue - should we file a new one?